lib/rouge/guessers/disambiguation.rb

```diff
@@ -90,14 +90,24 @@ def match?(filename)
       disambiguate '*.m' do
         next ObjectiveC if matches?(/@(end|implementation|protocol|property)\b/)
         next ObjectiveC if contains?('@"')
+        
+        # Objective-C dereferenced pointers and Mathematica comments are similar.
+        # Disambiguate for Mathematica by looking for any amount of whitespace (or no whitespace)
+        # followed by "(*" (e.g. `(* comment *)`).
+        next Mathematica if matches?(/^\s*\(\*/)
+
+        # Disambiguate for objc by looking for a deref'd pointer in a statement (e.g. `if (*foo == 0)`).
+        # This pattern is less specific than the Mathematica pattern, so its positioned after it.
+        next ObjectiveC if matches?(/^\s*(if|while|for|switch|do)\s*\([^)]*\*[^)]*\)/)
 
-        next Mathematica if contains?('(*')
         next Mathematica if contains?(':=')
 
         next Mason if matches?(/<%(def|method|text|doc|args|flags|attr|init|once|shared|perl|cleanup|filter)([^>]*)(>)/)
 
         next Matlab if matches?(/^\s*?%/)
-
+        # Matlab cell array creation: data = {
+        next Matlab if matches?(/^\s*[a-zA-Z]\w*\s*=\s*\{/)
```

Comment:
On the first round, I resolved an ambiguous match conflict by associating open curly braces with Objective-C. While this fixed the initial issue, it created a new conflict with Matlab. The revised Objective-C pattern now focuses on statements (if/while/for) and no longer requires the curly brace rule that conflicted with Matlab.

However, since it was mentioned, I still wanted to include a Matlab testcase involving curly braces. That revealed yet another ambiguous match scenario, which this rule addresses